lib/web_ui/lib/src/engine/skwasm/skwasm_impl/canvas.dart

```diff
@@ -259,21 +259,47 @@ class SkwasmCanvas implements SceneCanvas {
 
   @override
   void drawPoints(
-      ui.PointMode pointMode, List<ui.Offset> points, ui.Paint paint) {
-    throw UnimplementedError();
-  }
+    ui.PointMode pointMode,
+    List<ui.Offset> points,
+    ui.Paint paint
+  ) => withStackScope((StackScope scope) {
+    final RawPointArray rawPoints = scope.convertPointArrayToNative(points);
+    canvasDrawPoints(
+      _handle,
+      pointMode.index,
+      rawPoints,
+      points.length,
+      (paint as SkwasmPaint).handle,
+    );
+  });
 
   @override
   void drawRawPoints(
-      ui.PointMode pointMode, Float32List points, ui.Paint paint) {
-    throw UnimplementedError();
-  }
+    ui.PointMode pointMode,
+    Float32List points,
+    ui.Paint paint
+  ) => withStackScope((StackScope scope) {
+    final RawPointArray rawPoints = scope.convertDoublesToNative(points);
```

Comment:
Strange that we have to copy point data so many times. I imagine they are just uploaded to the GPU in a buffer...

Reply:
I think it's sort of the nature of the `drawPoints` API, since we're passing an ephemeral buffer of points. I guess there is sort of an extra copy going on too because skia also takes a const buffer of the points, so it's doing another copy under the hood. It's not likely to be that big of a deal, I think, and if users want to avoid the copies on each invocation of the API they should just use `drawVertices` instead and make the vertices object a single time.